src/main/java/macrobase/analysis/pipeline/OneShotPipeline.java

```diff
@@ -0,0 +1,36 @@
+package macrobase.analysis.pipeline;
+
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+
+abstract public class OneShotPipeline extends AbstractPipeline {
```

Comment:
> Seems more malleable then the iterator's next approach.

I agree it'd be convenient, but the goal is to make this iterators all the way down, independent of internal batching, etc.